src/Pages/ContentListTable/components/SnapshotDetailsModal/Tabs/SnapshotPackagesTab.tsx

```diff
@@ -0,0 +1,216 @@
+
+  const sortString = useMemo(
+    () => columnSortAttributes[activeSortIndex] + ':' + activeSortDirection,
+    [activeSortIndex, activeSortDirection],
+  );
+
+  const debouncedSearchQuery = useDebounce(searchQuery);
+
+  useEffect(() => {
+    setPage(1);
+  }, [debouncedSearchQuery, sortString]);
+
+  const {
+    isLoading,
+    isFetching,
+    isError,
+    data = { data: [], meta: { count: 0, limit: 20, offset: 0 } },
+  } = useGetSnapshotPackagesQuery(snapshotUUID, page, perPage, debouncedSearchQuery, sortString);
+
+  useEffect(() => {
+    if (isError) {
+      onClose();
+    }
+  }, [isError]);
+
+  const onSetPage = (_, newPage) => setPage(newPage);
+
+  const onPerPageSelect = (_, newPerPage, newPage) => {
+    setPerPage(newPerPage);
+    setPage(newPage);
+    localStorage.setItem(perPageKey, newPerPage.toString());
+  };
+
+  const sortParams = (columnIndex: number, isDisabled: boolean): ThProps['sort'] | undefined => {
+    if (isDisabled) return;
+    return {
+      sortBy: {
+        index: activeSortIndex,
+        direction: activeSortDirection,
+        defaultDirection: 'asc',
+      },
+      onSort: (_event, index, direction) => {
+        setActiveSortIndex(index);
+        setActiveSortDirection(direction);
+      },
+      columnIndex,
+    };
+  };
+
+  const onClose = () =>
+    navigate(rootPath + (contentOrigin === ContentOrigin.REDHAT ? `?origin=${contentOrigin}` : ''));
+
+  const {
+    data: packageList = [],
+    meta: { count = 0 },
+  } = data;
+
+  const fetchingOrLoading = isFetching || isLoading;
+
+  const loadingOrZeroCount = fetchingOrLoading || !count;
+  return (
+    <Grid className={classes.mainContainer}>
+      <InputGroup className={classes.topContainer}>
+        <InputGroupItem>
+          <TextInput
+            id='search'
+            ouiaId='name_search'
+            placeholder='Filter by name'
+            value={searchQuery}
+            onChange={(_event, value) => setSearchQuery(value)}
+          />
+          <InputGroupText id='search-icon'>
+            <SearchIcon />
+          </InputGroupText>
+        </InputGroupItem>
+        <Hide hide={isLoading}>
+          <Pagination
+            id='top-pagination-id'
+            widgetId='topPaginationWidgetId'
+            itemCount={count}
+            perPage={perPage}
+            page={page}
+            onSetPage={onSetPage}
+            isCompact
+            onPerPageSelect={onPerPageSelect}
+          />
+        </Hide>
+      </InputGroup>
+      <Hide hide={!fetchingOrLoading}>
+        <Grid className={classes.mainContainer}>
+          <SkeletonTable
+            rows={perPage}
+            numberOfColumns={columnHeaders.length}
+            variant={TableVariant.compact}
+          />
+        </Grid>
+      </Hide>
+      <Hide hide={fetchingOrLoading}>
+        <Table aria-label='Custom repositories table' ouiaId='packages_table' variant='compact'>
```

Comment:
i think this aria-label should be `snapshot_packages_table` or something similar